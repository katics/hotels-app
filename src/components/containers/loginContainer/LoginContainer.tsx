import { ChangeEvent, FC, useEffect, useState } from "react";

import store from "../../../store";
import HotelProvider, {
  hotelContext,
  useDispatchHotel,
  useSelectorHotel,
} from "../../../store/HotelProvider";
import { fetchLoginUser } from "../../../store/user/userActions";
import { userSeletor } from "../../../store/user/userSelector";
import { APP_ROUTES } from "../../../utils/AppRoutes";
import { UserLoginData } from "../../../utils/types/UserLoginData";
import { useHistory } from "react-router";
import LoginPage from "../../pages/loginPage/LoginPage";
import { ILoginProps } from "../../../utils/props/LoginPageProps";
import { spinnerSelector } from "../../../store/spinner/spinnerSelector";
import Spinner from "../../spinner/Spinner";

const LoginContainer: FC = () => (
  <HotelProvider store={store} context={hotelContext}>
    <UserLogin />
  </HotelProvider>
);

const UserLogin: FC = () => {
  const history = useHistory();

  const userDispatch = useDispatchHotel();
  const { loginError, isLogged, isLoading } = useSelectorHotel(userSeletor);
  //const { isLoading } = useSelectorHotel(spinnerSelector);
  const userLoginData: UserLoginData = { username: "", password: "" };
  const [loginData, setLoginData] = useState(userLoginData);

  const [spinner, setSpinner] = useState(isLoading);

  const setUserName = (e: ChangeEvent<HTMLInputElement>) => {
    setLoginData({
      ...loginData,
      username: e.target.value,
    });
  };
  const setPassword = (e: ChangeEvent<HTMLInputElement>) => {
    setLoginData({
      ...loginData,
      password: e.target.value,
    });
  };

  const signInClick = (e: ChangeEvent<HTMLFormElement>) => {
    e.preventDefault();
    userDispatch(fetchLoginUser(loginData));
  };

  useEffect(() => {
    console.log("Redirect na loginu, user je logovan: " + isLogged.toString());
    if (isLogged) {
      history.push(APP_ROUTES.dashboard);
    }
  }, [isLogged]);

  const loginProps: ILoginProps = {
    loginData: loginData,
    loginError: loginError,
    signInClick: signInClick,
    setUserName: setUserName,
    setPassword: setPassword,
  };
  return <>{isLoading ? <Spinner /> : <LoginPage loginProps={loginProps} />}</>;
};

export default LoginContainer;
