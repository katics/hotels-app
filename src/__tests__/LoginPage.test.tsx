import LoginContainer from "../components/containers/loginContainer/LoginContainer";
import { StaticRouter } from "react-router-dom";

import { render } from "@testing-library/react";

describe("Login Page", () => {
  it("Should match login page snapshot", () => {
    const { container } = render(
      <StaticRouter>
        <LoginContainer />
      </StaticRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
