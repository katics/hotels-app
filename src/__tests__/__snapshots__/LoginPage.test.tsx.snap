// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page Should match login page snapshot 1`] = `
<div
  class="col-lg-6 login-container"
>
  <div
    class="card login-form"
  >
    <div
      class="card-body"
    >
      <h3
        class="card-title text-center"
      >
        Log in
      </h3>
      <div
        class="card-text"
      >
        <form>
          <div
            class="form-group"
          >
            <label>
              User Name
            </label>
            <input
              class="form-control"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label>
              Password
            </label>
            <input
              class="form-control"
              type="password"
              value=""
            />
          </div>
          <button
            class="btn btn-primary btn-block"
            type="submit"
          >
            Sign in
          </button>
          <div
            class="sign-up"
          >
            Don't have an account?
            <a
              class="float-right"
              href="/register"
            >
              Register
            </a>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
